app/policies/post_policy.rb

```diff
@@ -1,5 +1,20 @@
 class PostPolicy < ApplicationPolicy
+  class Scope < Scope
+    def resolve
+      if user.present?
+        if user.admin? || user.moderator?
+          scope.all
+        else  
+          scope.where(:user_id => user.id)
```

Comment:
When you have a symbol as the key of a hash, most style guides recommend using the newer `user_id: user.id` syntax.
See [this](https://github.com/bbatsov/ruby-style-guide) for reference